### PR TITLE
Adds token validation on handleAuthResponse

### PR DESCRIPTION
**Pull request by eluciano11**
Adds token validation on handleAuthResponse. Users should not get an error Uncaught TypeError: Cannot read property 'split' of undefined

### CODE REVIEW

**Comment by jpadilla** [comment]
I was thinking we could do something like this. We'd warn about empty token, and prevent the uncaught typerror. Thoughts?

``` javascript
getTokenData(token) {
  if (Ember.isEmpty(token)) {
    Ember.Logger.warn('TOKEN IS EMPTY OR A BETTER WARNING');

    return {};
  }

  const payload = token.split('.')[1];
  const tokenData = decodeURIComponent(window.escape(atob(payload)));

  try {
    return JSON.parse(tokenData);
  } catch (e) {
    return tokenData;
  }
},
```

**Comment by eluciano11** [comment]
Yeah, I thought about managing it in the `getTokenData` function but the `restore` function was using something [similar](https://github.com/jpadilla/ember-simple-auth-token/pull/147/files#diff-b3b19e97b14851cfabb1e6558abfc6daL110) to what I implemented on `handleAuthResponse`. We could remove that validation from `restore` and manage it in `getTokenData` as suggested above.

File: addon/authenticators/jwt.js [hunk 1]
```diff
@@ -220,9 +226,14 @@ export default TokenAuthenticator.extend({
...
-          const sessionData = this.handleAuthResponse(response);
-          this.trigger('sessionDataUpdated', sessionData);
-          resolve(sessionData);
+          try {
+            const sessionData = this.handleAuthResponse(response);
+
+            this.trigger('sessionDataUpdated', sessionData);
+            resolve(sessionData);
+          } catch(error) {
+            reject(new Error(error));
```
Comment (jpadilla) [inline addon/authenticators/jwt.js]: Should be `reject(error);`

File: addon/authenticators/jwt.js [hunk 2]
```diff
@@ -163,8 +164,13 @@ export default TokenAuthenticator.extend({
...
       this.makeRequest(this.serverTokenEndpoint, data, headers).then(response => {
         Ember.run(() => {
-          const sessionData = this.handleAuthResponse(response);
-          resolve(sessionData);
+          try {
+            const sessionData = this.handleAuthResponse(response);
+
+            resolve(sessionData);
+          } catch(error) {
+            reject(new Error(error));
```
Comment (jpadilla) [inline addon/authenticators/jwt.js]: Should be `reject(error);`

**Comment by eluciano11** [comment]
@homu r+

**Pull request merged**